src/utils/request/onUnhandledRequest.ts

```diff
@@ -46,18 +48,23 @@ function groupHandlersByType(handlers: RequestHandler[]) {
   )
 }
 
-type RequestHandlerSuggestionList = [number, RequestHandler][]
-type ScoreGetterFn = (request: MockedRequest, handler: RequestHandler) => number
+type RequestHandlerSuggestion = [number, RequestHandler]
 
-function getScoreForRestHandler(): ScoreGetterFn {
+type ScoreGetterFn<RequestHandlerType extends RequestHandler> = (
+  request: MockedRequest,
+  handler: RequestHandlerType,
+) => number
+
+function getRestHandlerScore(): ScoreGetterFn<RestHandler> {
   return (request, handler) => {
     const { path, method } = handler.info
 
-    if (path instanceof RegExp) {
+    if (path instanceof RegExp || method instanceof RegExp) {
```

Comment:
I haven't yet understood the concept of _score_ in the project so, sorry for asking this, but assuming that this is attributing a _priority_ for each handler, won't this hurt in cases like:

```js
rest.get('/one/endpoint', () => { /* ... does something ... */ });
rest.all('/one/endpoint', () => { /* ... does something different ... */ });
```

If you are certain that the method's RegExp matches with the current request's method, then you can ignore the method here, no?

Reply:
Yes, I think the `rest.all()` handler must not be suggested. Please see my reasoning around this below.